### ISSUE TITLE
Use template values for Ingress annotations 

### ISSUE THREAD

**Issue by rjminsha** [helm/monocular#288]
Ingress may vary across environments.  I think it would be good to remove 
```
    ingress.kubernetes.io/rewrite-target: /
```
from `templates/ingress.yaml` and relying upon values in `values.yaml`.  This will allow deployment with different ingress implementations to simply provide a values.yaml rather than updating the templates themselves.

**Comment by prydonius** [comment]
Whilst I agree that it would be good to template this, a few controllers are missing support for this which will cause them not to work. The API is designed to listen to requests at the root (e.g. `/v1/charts` not `/api/v1/charts` so if these are not correctly rewritten before forwarding the request, it won't work.

This has caused issues in the past, so we've taken to saying that we only really support the nginx ingress controller out of the box, and otherwise you can manually configure your own Ingress or use separate domains as described [here](https://github.com/helm/monocular/blob/master/deployment/monocular/README.md#serve-monocular-frontend-and-api-on-different-domains).

**Comment by rjminsha** [comment]
@prydonius by controllers I guess you mean ingress controllers.  The nginx controller does support the rewriting and others take similar but different formats.  

What I'm proposing is not that all ingress controllers are supported, but rather the annotation for nginx be moved to the values.yaml so that the annotation can be overridden is someone is using another ingress that supports rewriting but with a different annotation. While recognizing the need for a rewrite I suppose I'm just missing why that particular annotation is in the Ingress directly. 

Something like 
``` 
ingress:
  enabled: true
  annotations:
    ingress.kubernetes.io/rewrite-target: /
```

ingress
``` 
  annotations:
  {{- range $key, $value := .Values.ingress.annotations }}
    {{ $key }}: {{ $value | quote }}
  {{- end }}
```  
If I were using a different ingress I could rely on values.yaml rather than forking the monocolar repo.

**Comment by prydonius** [comment]
@rjminsha I've updated the chart in #297 to move these annotations to values.yaml. Can you give that a quick review?

**Comment by rjminsha** [comment]
Perfect.  Thanks.

**Comment by prydonius** [comment]
chart version includes the fix 0.4.4 :)

**Comment by mistyle** [comment]
I met with you exactly the same problem, how will the final solution?
